osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeIntro.cs

```diff
@@ -93,14 +93,15 @@ public DailyChallengeIntro(Room room)
         protected override BackgroundScreen CreateBackground() => new DailyChallengeIntroBackgroundScreen(colourProvider);
 
         [BackgroundDependencyLoader]
-        private void load(BeatmapDifficultyCache difficultyCache, BeatmapModelDownloader beatmapDownloader, OsuConfigManager config, AudioManager audio)
+        private void load(RulesetStore rulesets, BeatmapDifficultyCache difficultyCache, BeatmapModelDownloader beatmapDownloader, OsuConfigManager config, AudioManager audio)
         {
             const float horizontal_info_size = 500f;
 
-            Ruleset ruleset = Ruleset.Value.CreateInstance();
-
             StarRatingDisplay starRatingDisplay;
 
+            IBeatmapInfo beatmap = item.Beatmap;
+            Ruleset ruleset = rulesets.GetRuleset(item.Beatmap.Ruleset.ShortName)?.CreateInstance() ?? Ruleset.Value.CreateInstance();
```

Comment:
Why does this still fallback to `Ruleset.Value.CreateInstance()`? I don't see the point, it can literally be only more wrong than the left half of this.

Reply:
Fair point.

Reply:
Do you think the mods just shouldn't show in this case? Because I think that would be the closest alternative (we need a ruleset to show the `??` mods).

Reply:
I'd just delete the fallback here and be done with it.

Reply:
Right. I guess it's always going to be an official ruleset anyway.